File: Java_Codes/8. IO Streams/BufferedReaderExample.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class BufferedReaderExample {
    public static void readFileLineByLine(String filePath) {
        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
            String line;
            while ((line = br.readLine()) != null) {
                System.out.println(line);
            }
        } catch (IOException e) {
            System.out.println("Error reading file: " + e.getMessage());
        }
    }

    public static void main(String[] args) {
        readFileLineByLine("C:\\Java_Codes\\8. IO Streams\\example2.txt");
    }
}

File: Python_Codes/8. IO Streams/BufferedReaderExample.py
def read_file_line_by_line(file_path):
    try:
        with open(file_path, 'r') as file:
            for line in file:
                print(line.strip())  # strip() removes the newline character
    except IOError as e:
        print("Error reading file:", e)


if __name__ == "__main__":
    read_file_line_by_line("C:/WorkSpace/Python_Codes/8. IO Streams/example2.txt")
